File: server/DataCore/invg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace inv19.models { 
	/* invg -  Группы */ 

 public class  invg_grp { // Группа
	 public System.Guid  invg_grpId{ get; set; } // Идентификатор (первичный ключ)
	 public List<invg_subgrp>  invg_subgrp { get; set; } // дочерний раздел: Подгруппа
	[Required]
	public string  name{ get; set; } // Название
 }

 public class  invg_subgrp { // Подгруппа
	 public System.Guid  invg_subgrpId{ get; set; } // Идентификатор (первичный ключ)
	[Required]
	 public System.Guid  invg_grpId { get; set; } // обратная ссылка на: Группа
	[Required]
	public string  name{ get; set; } // Название
 }
}

File: client/src/app/invg.ts
import { enums } from './enums';

export namespace invg { 
	/* invg -  Группы */ 

 export interface   invg_grp { // Группа
	invg_grpId:string; // Primary key field
	name:string; // Название
 }

 export interface   invg_subgrp { // Подгруппа
	invg_subgrpId:string; // Primary key field
	  invg_grpId:string; // Группа
	name:string; // Название
 }
}